src/zope/datetime/__init__.py

```diff
@@ -17,281 +17,297 @@
-    StringTypes = (str, )  # python3 and up
+from datetime import datetime as _datetime
+from datetime import tzinfo as _std_tzinfo, timedelta as _timedelta
+
+
+from zope.datetime.timezones import historical_zone_info as _data
 
-try:
-    from time import tzname
-except ImportError:
-    tzname = ('UNKNOWN', 'UNKNOWN')
+
+StringTypes = (str,) if str is not bytes else (basestring,)
 
 
 # These are needed because the various date formats below must
 # be in english per the RFCs. That means we can't use strftime,
 # which is affected by different locale settings.
-weekday_abbr = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']
-weekday_full = ['Monday', 'Tuesday', 'Wednesday', 'Thursday',
-                'Friday', 'Saturday', 'Sunday']
-monthname    = [None, 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
-                'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
+weekday_abbr = [
+    'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun',
+]
+weekday_full = [
+    'Monday', 'Tuesday', 'Wednesday', 'Thursday',
+    'Friday', 'Saturday', 'Sunday',
+]
+monthname = [
+    None, 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
+    'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec',
+]
 
 
 def iso8601_date(ts=None):
     # Return an ISO 8601 formatted date string, required
     # for certain DAV properties.
     # '2000-11-10T16:21:09-08:00
-    if ts is None:
-        ts = _time.time()
+    ts = _time.time() if ts is None else ts
     return _time.strftime('%Y-%m-%dT%H:%M:%SZ', _time.gmtime(ts))
 
 def rfc850_date(ts=None):
     # Return an HTTP-date formatted date string.
     # 'Friday, 10-Nov-00 16:21:09 GMT'
-    if ts is None:
-        ts = _time.time()
-    year, month, day, hh, mm, ss, wd, y, z = _time.gmtime(ts)
+    ts = _time.time() if ts is None else ts
+    year, month, day, hh, mm, ss, wd, _y, _z = _time.gmtime(ts)
     return "%s, %02d-%3s-%2s %02d:%02d:%02d GMT" % (
-            weekday_full[wd],
-            day, monthname[month],
-            str(year)[2:],
-            hh, mm, ss)
+        weekday_full[wd],
+        day, monthname[month],
+        str(year)[2:],
+        hh, mm, ss
+    )
 
 def rfc1123_date(ts=None):
     # Return an RFC 1123 format date string, required for
     # use in HTTP Date headers per the HTTP 1.1 spec.
     # 'Fri, 10 Nov 2000 16:21:09 GMT'
-    if ts is None:
-        ts = _time.time()
-    year, month, day, hh, mm, ss, wd, y, z = _time.gmtime(ts)
-    return "%s, %02d %3s %4d %02d:%02d:%02d GMT" % (weekday_abbr[wd],
-                                                    day, monthname[month],
-                                                    year,
-                                                    hh, mm, ss)
+    ts = _time.time() if ts is None else ts
+    year, month, day, hh, mm, ss, wd, _y, _z = _time.gmtime(ts)
+    return "%s, %02d %3s %4d %02d:%02d:%02d GMT" % (
+        weekday_abbr[wd],
+        day, monthname[month],
+        year,
+        hh, mm, ss)
+
+class DateTimeError(Exception):
+    "Date-time error"
```

Comment:
PEP-8 thinks there should be two newlines between the class definitions.
PEP-257 wants to have triple quotes.